raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -149,9 +149,12 @@ def encode_function_call(self, function: str, args: List = None):
     def estimate_gas(
         self, block_identifier, function: str, *args, **kwargs
     ) -> typing.Optional[int]:
-        """Returns a gas estimate for the function with the given arguments or
-        None if the function call will fail due to Insufficient funds or
-        the logic in the called function."""
+        """Estimate the gas necessary to run the transaction.
+
+        Returns `None` transaction would fail because it hit an assert/revert,
```

Comment:
... "because it hit an assert, revert or require"

Also, why do we just test for `assert` and `require` and not for `revert` as well?

Reply:
AFAIK there is no `revert` keyword. The transaction is reverted when a `require`d condition is not met, so the `require` should cover it.

Reply:
https://solidity.readthedocs.io/en/v0.5.10/control-structures.html#revert

Reply:
HUmm, I didn't know about `revert`. Is it useful to add it though?

> The two syntax options are equivalent, it’s developer preference which to use.

Reply:
I'd say let's cover `revert` too, it may be used in our contracts at some point... But we can also merge like this and add it later.

Reply:
Yes, my point is that the behavior of a `revert` and a `require` is identical, it is just syntatic sugar for the `revert` opcode.